src/app/dynamic/enterprise/metering/legacy-reports/list/template.html

```diff
@@ -0,0 +1,100 @@
+
+1. You may only view, read and display for studying purposes the source
+   code of the software licensed under this license, and, to the extent
+   explicitly provided under this license, the binary code.
+2. Any use of the software which exceeds the foregoing right, including,
+   without limitation, its execution, compilation, copying, modification
+   and distribution, is expressly prohibited.
+3. THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+   IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+   CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+   TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+   SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+END OF TERMS AND CONDITIONS
+-->
+
+<table class="km-table"
+       mat-table
+       matSort
+       matSortActive="name"
+       matSortDirection="asc"
+       [dataSource]="dataSource">
+  <ng-container [matColumnDef]="column.name">
+    <th mat-header-cell
+        *matHeaderCellDef
+        class="km-header-cell p-60"
+        mat-sort-header>Name</th>
+    <td mat-cell
+        *matCellDef="let element">{{element.name}}</td>
+  </ng-container>
+
+  <ng-container [matColumnDef]="column.size">
+    <th mat-header-cell
+        *matHeaderCellDef
+        class="km-header-cell p-20"
+        mat-sort-header>Size</th>
+    <td mat-cell
+        *matCellDef="let element">{{element.size | kmSize}}</td>
+  </ng-container>
+
+  <ng-container [matColumnDef]="column.lastModified">
+    <th mat-header-cell
+        *matHeaderCellDef
+        class="km-header-cell p-20"
+        mat-sort-header>Created</th>
+    <td mat-cell
+        *matCellDef="let element">
+      <km-relative-time [date]="element.lastModified"></km-relative-time>
+    </td>
+  </ng-container>
+
+  <ng-container [matColumnDef]="column.actions">
+    <th mat-header-cell
+        *matHeaderCellDef
+        class="km-header-cell"></th>
+    <td mat-cell
+        *matCellDef="let element">
+      <div class="km-table-actions"
+           fxLayoutAlign="end">
+        <ng-container *ngIf="isProcessingReport(element)">
+          <mat-spinner color="accent"
+                       class="km-spinner"
+                       fxFlexAlign=" center"
+                       [diameter]="25"></mat-spinner>
+        </ng-container>
+
+        <button id="km-download-report-button"
+                mat-icon-button
+                kmThrottleClick
+                (throttleClick)="download(element.name)">
+          <i class="km-icon-mask km-icon-download"></i>
+        </button>
+
+        <button mat-icon-button
+                kmThrottleClick
+                (throttleClick)="remove(element.name)">
+          <i class="km-icon-mask km-icon-delete"></i>
+        </button>
```

Comment:
I think we should hide action buttons when spinner is displayed.

```suggestion
        <ng-container *ngIf="isProcessingReport(element); else actionButtons">
          <mat-spinner color="accent"
                       class="km-spinner"
                       fxFlexAlign=" center"
                       [diameter]="25"></mat-spinner>
        </ng-container>
        
        <ng-template #actionButtons>
          <button id="km-download-report-button"
                  mat-icon-button
                  kmThrottleClick
                  (throttleClick)="download(element.name)">
            <i class="km-icon-mask km-icon-download"></i>
          </button>

          <button mat-icon-button
                  kmThrottleClick
                  (throttleClick)="remove(element.name)">
            <i class="km-icon-mask km-icon-delete"></i>
          </button>
        </ng-template>
```